partiql-eval/src/main/kotlin/org/partiql/eval/internal/Compiler.kt

```diff
@@ -220,13 +221,37 @@ internal class Compiler(
     @OptIn(FnExperimental::class)
     override fun visitRexOpCallDynamic(node: Rex.Op.Call.Dynamic, ctx: StaticType?): Operator {
         val args = node.args.map { visitRex(it, ctx).modeHandled() }.toTypedArray()
+        // Check candidate list size
+        when (node.candidates.size) {
+            0 -> error("Rex.Op.Call.Dynamic had an empty candidates list: $node.")
+            // TODO this seems like it should be an error, but is possible if the fn match was non-exhaustive
+            // 1 -> error("Rex.Op.Call.Dynamic had a single candidate; should be Rex.Op.Call.Static")
+        }
+        // Check candidate name and arity for uniformity
+        var arity: Int = -1
+        var name: String = "unknown"
+        // Compile the candidates
         val candidates = Array(node.candidates.size) {
             val candidate = node.candidates[it]
             val fn = symbols.getFn(candidate.fn)
             val coercions = candidate.coercions.toTypedArray()
+            // Check this candidate
+            val fnArity = fn.signature.parameters.size
+            val fnName = fn.signature.name
+            if (arity == -1) {
+                arity = fnArity
+                name = fnName
+            } else {
+                if (fnArity != arity) {
+                    error("Dynamic call candidate had different arity than others; found $fnArity but expected $arity")
+                }
+                if (fnName != name) {
+                    error("Dynamic call candidate had different name than others; found $fnName but expected $name")
+                }
+            }
```

Comment:
Is it necessary to perform these checks?

Also, we shouldn't be checking the name this way due to case-insensitive lookup. If we need these checks, then I'd consider making `var name` maintain a normalized function name.

Reply:
Yes, as it's basic one-time input validation. Not strictly necessary if plans come from the internal partiql-planner — but that might not always be where we receive plans from. 

Ack on the normalized names.